lib/Dialect/FIRRTL/FIRRTLFolds.cpp

```diff
@@ -237,30 +326,51 @@ OpFoldResult AddPrimOp::fold(ArrayRef<Attribute> operands) {
                                  [=](APSInt a, APSInt b) { return a + b; });
 }
 
-void AddPrimOp::getCanonicalizationPatterns(RewritePatternSet &results,
-                                            MLIRContext *context) {
-  results.insert<patterns::AddWithInvalidOp>(context);
+LogicalResult AddPrimOp::canonicalize(AddPrimOp op, PatternRewriter &rewriter) {
+  return canonicalizePrimOp(op, rewriter,
+                            [&](ArrayRef<Attribute> operands) -> OpFoldResult {
+                              // add(x, 0) -> x
+                              // add(0, x) -> x
+                              if (isConstantZero(operands[1]))
+                                return op.getOperand(0);
+                              if (isConstantZero(operands[0]))
+                                return op.getOperand(1);
+                              return {};
+                            });
```

Comment:
Nit: it may be more readable to move the comment onto the `if` statement it correlates with as opposed to putting them all at the top. Ditto for below.

Reply:
Yeah good thinking.